### ISSUE TITLE
False positive on manual impl of Copy and Clone

### ISSUE THREAD

**Issue by llogiq** [rust-lang/rust-clippy#893]
From [petgraph](https://github.com/bluss/petgraph):

```
impl<'b, T> Copy for Ptr<'b, T> {}
impl<'b, T> Clone for Ptr<'b, T>
{
    fn clone(&self) -> Self { *self }
}
```

Note that this expressly omits `Copy`/`Clone` bounds for `T`, respectively.

**Comment by oli-obk** [comment]
rustc now allows

```rust
#[derive(Copy, Clone)]
struct Ptr<'b, T: 'b>(&'b T);
```

so we can close this?

**Comment by camsteffen** [comment]
This can be an enhancement of `derivable_impls`

**Comment by Enselic** [comment]
I can't reproduce this false positive. The latest code from petgraph is [here](https://github.com/petgraph/petgraph/blob/9ff688872b467d3e1b5adef19f5c52f519d3279c/src/graphmap.rs#L834-L841) and it matches the [original code](https://github.com/rust-lang/rust-clippy/issues/893#issue-152666088) for this issue. But clippy from latest stable Rust 1.57 does not complain:

```
% cat src/lib.rs
pub struct Ptr<'b, T: 'b>(pub &'b T);

impl<'b, T> Copy for Ptr<'b, T> {}
impl<'b, T> Clone for Ptr<'b, T> {
    fn clone(&self) -> Self {
        *self
    }
}

% cargo clippy -- --deny clippy::all --deny clippy::pedantic
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s

% cargo clippy -- --version
clippy 0.1.57 (f1edd042 2021-11-29)
```

So I think we can close this issue?
